app/gui2/src/appRunner.ts

```diff
@@ -11,6 +11,14 @@ async function runApp(
   _metadata?: object | undefined,
   pinia?: Pinia | undefined,
 ) {
+  const ignoreParamsRegex = (() => {
+    if (_metadata)
+      if ('ignoreParamsRegex' in _metadata)
+        if (_metadata['ignoreParamsRegex'] instanceof RegExp) return _metadata['ignoreParamsRegex']
+
+    return null
+  })()
```

Comment:
I use the self-invoke function here because I found it more readable than ternary:
```typescript
const ignoreParamsRegex = metadata && 'ignoreParamsRegex' in _metadata &&  _metadata['ignoreParamsRegex'] instanceof RegExp ? _metadata['ignoreParamsRegex'] : null
```

Reply:
> I use the self-invoke function here because I found it more readable than the ternary:

Formatted, it does not look so bad IMO. The ifs are a bit noisy as for something not being a primary topic of this function.
```ts
  const ignoreParamsRegex =
    metadata && 'ignoreParamsRegex' in metadata && metadata['ignoreParamsRegex'] instanceof RegExp ?
      metadata['ignoreParamsRegex']
    : null
```